### PR TITLE
DEV: CI job for build and run dpnp backend clib tests

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
* removed building tests from `devenv_lin` job
* added new job for building and running DPNP backend clib tests
  *  building without python

### CODE REVIEW

**Comment by samir-nasibli** [comment]
@shssf I can make a new .sh in `srcipts` dir for this build and run or just leave it as is on `scripts/azure-pipelines.yml`. Which one is more preferable?

**Comment by samir-nasibli** [comment]
@shssf currenlty there are some test fails due to lack of Intel GPU on our azure machines. Fixes on #552

**Comment by Alexander-Makaryev** [comment]
test_dpnp_backend_clib_lin (added in this PR) has failed status, is it ok?

**Comment by samir-nasibli** [comment]
> test_dpnp_backend_clib_lin (added in this PR) has failed status, is it ok?

Yes, it is ok. See, prev comment, please. https://github.com/IntelPython/dpnp/pull/559#issuecomment-772030190

**Comment by samir-nasibli** [comment]
> > test_dpnp_backend_clib_lin (added in this PR) has failed status, is it ok?
> 
> Yes, it is ok. See, prev comment, please. [#559 (comment)](https://github.com/IntelPython/dpnp/pull/559#issuecomment-772030190)

We need merge #552 before, if we want green ci for this job.

File: scripts/azure-pipelines.yml [hunk 1]
```diff
@@ -311,5 +311,29 @@ jobs:
...
+      echo ========================= build DPNP package ===============================
+      export DPNP_BACKEND_TESTS_ENABLE=1
+      . ./0.env.sh
+      THEDIR=$(pwd)
+      cmake  -S$THEDIR/dpnp/backend -B$THEDIR/build/ -DCMAKE_BUILD_TYPE=Release \
+              -DDPNP_INSTALL_PREFIX=$THEDIR/dpnp -DDPNP_INSTALL_STRUCTURED=OFF \
+              -DDPNP_SYCL_QUEUE_MGR_ENABLE:BOOL=OFF \
+              -DDPNP_QUEUEMGR_INCLUDE_DIR=No_sycl_queue_mgr_include_dir \
+              -DDPNP_QUEUEMGR_LIB_DIR=No_sycl_queue_mgr_library_dir \
+              -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DDPNP_BACKEND_TESTS:BOOL=ON \
```
Comment (shssf) [inline scripts/azure-pipelines.yml:334]: I would propose to do not use all these variables here (like `DPNP_QUEUEMGR_INCLUDE_DIR`). Most of these variables have default values that should be used here (just do not put them here).

File: scripts/azure-pipelines.yml [hunk 2]
```diff
@@ -311,5 +311,29 @@ jobs:
...
+  displayName: test_dpnp_backend_clib_lin
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+  - bash: |
+      . ./scripts/install_cmake_lin.sh
+      ./scripts/install_system_deps.sh  # no intel python
+      echo ========================= build DPNP package ===============================
+      export DPNP_BACKEND_TESTS_ENABLE=1
+      . ./0.env.sh
```
Comment (shssf) [inline scripts/azure-pipelines.yml:327]: This script is from "python" part. I think it is better to use ". /opt/intel/oneapi/setvars.sh" instead.

File: scripts/azure-pipelines.yml [hunk 3]
```diff
@@ -311,5 +311,29 @@ jobs:
...
+              -DDPNP_INSTALL_PREFIX=$THEDIR/dpnp -DDPNP_INSTALL_STRUCTURED=OFF \
+              -DDPNP_SYCL_QUEUE_MGR_ENABLE:BOOL=OFF \
+              -DDPNP_QUEUEMGR_INCLUDE_DIR=No_sycl_queue_mgr_include_dir \
+              -DDPNP_QUEUEMGR_LIB_DIR=No_sycl_queue_mgr_library_dir \
+              -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DDPNP_BACKEND_TESTS:BOOL=ON \
+              $THEDIR/dpnp/backend
+      cmake --build $THEDIR/build
+      cmake --install $THEDIR/build
+      echo ======================= run DPNP backend tests =============================
+      ./build/tests/dpnpc_tests
```
Comment (shssf) [inline scripts/azure-pipelines.yml:339]: it looks like tests binary is not properly installed by `make install` procedure. It looks like it needs to be implemented in CMakeFiles.txt

File: scripts/azure-pipelines.yml [hunk 4]
```diff
@@ -311,5 +311,26 @@ jobs:
...
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+  - bash: |
+      . ./scripts/install_cmake_lin.sh
+      ./scripts/install_system_deps.sh  # no intel python
+      echo ========================= build DPNP package ===============================
+      . /opt/intel/oneapi/setvars.sh
+      THEDIR=$(pwd)
+      cmake  -S$THEDIR/dpnp/backend -B$THEDIR/build/ -DCMAKE_BUILD_TYPE=Release \
```
Comment (shssf) [inline scripts/azure-pipelines.yml:328]: make it simpler. Do not use unnecessary variables like `CMAKE_BUILD_TYPE` or `CMAKE_VERBOSE_MAKEFILE` and etc

Comment (samir-nasibli) [reply scripts/azure-pipelines.yml:328]: This is not so critical. But I can remove it if you insist.

Comment (samir-nasibli) [reply scripts/azure-pipelines.yml:328]: This is not so critical. But I can remove it if you insist.

**Review by densmirn — APPROVED** [review]

**Review by shssf — APPROVED** [review]

**Review by shssf — APPROVED** [review]

**Pull request merged**